server/src/modules/links/helpers.py

```diff
@@ -198,9 +198,17 @@ def _is_valid_bare_hostname_destination(destination):
 
   return type(validators.url(with_tld)) is not ValidationFailure
 
+"""Returns whether or not the provided destination is a valid URL using double hyphen, like `https://double--hyphen.example.com/some/path`.
+
+This function is used to work around a limitation of the `validators` package where it doesn't recognize URLs with double hypens valid
+"""
+def _is_valid_idn_destination(destination):
+  clean_url = re.sub('\-\-+', '-', destination)
+  return validators.url(clean_url) is not ValidationFailure
```

Comment:
```suggestion
  return type(validators.url(clean_url)) is not ValidationFailure
```

`validators.url(clean_url) is not ValidationFailure` alone will always be true